core/trino-main/src/main/java/io/trino/execution/QueryManagerConfig.java

```diff
@@ -146,6 +146,7 @@ public class QueryManagerConfig
     private boolean faultTolerantExecutionSmallStageRequireNoMorePartitions;
     private boolean faultTolerantExecutionStageEstimationForEagerParentEnabled = true;
     private boolean faultTolerantExecutionMarkDistinctChainEstimationEnabled = true;
+    private int faultTolerantExecutionMarkDistinctChainEstimationMaxDepth = 5;
```

Comment:
>Why 5? Document how this value was picked

Hmm - there was not much of a though process here. 5 is not that much - to be adjusted if we see it does not work well.

Reply:
I put:
```
    // limit speculative execution of mark distinct chain length to arbitrary depth to avoid
    // excessive entropy in estimation result
```